core/io/dir_access.cpp

```diff
@@ -35,13 +35,20 @@
 #include "core/os/os.h"
 #include "core/os/time.h"
 #include "core/templates/local_vector.h"
+#ifdef TOOLS_ENABLED
+#include "editor/settings/editor_settings.h"
+#endif
 
 String DirAccess::_get_root_path() const {
 	switch (_access_type) {
 		case ACCESS_RESOURCES:
 			return ProjectSettings::get_singleton()->get_resource_path();
 		case ACCESS_USERDATA:
 			return OS::get_singleton()->get_user_data_dir();
+#ifdef TOOLS_ENABLED
```

Comment:
Having it editor only is questionable in general. Instead, I would:
- move `get_global_resource_path` to `ProjectSettings` and add (unexposed to scripting) `set_global_resource_path`.
- editor init can use it to set editor specific path, and if it is not set, it will return the same path as `get_resource_path`.